File: HRDashboard/HRDashboard/Models/Insight.cs
namespace HRDashboard.Models;

public class Insight
{
    public Guid Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public InsightType Type { get; set; }

    public InsightSeverity Severity { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: frontend/src/types/insight.ts
export type InsightType = 'Positive' | 'Negative' | 'Neutral' | 'Warning';

export type InsightSeverity = 'Info' | 'Warning' | 'Critical';

export interface InsightResponse {
  id: string;
  title: string;
  description: string | null;
  type: InsightType;
  severity: InsightSeverity;
  createdAt: string;
  updatedAt: string;
}

export interface CreateInsight {
  title: string;
  description: string | null;
  type: InsightType;
  severity: InsightSeverity;
}

export type UpdateInsight = CreateInsight;